Fabric.Identity.API/Services/ClaimsService.cs

```diff
@@ -0,0 +1,147 @@
+﻿using Fabric.Identity.API.Configuration;
+using Fabric.Identity.API.Exceptions;
+using Fabric.Identity.API.Models;
+using Fabric.Identity.API.Quickstart.Account;
+using IdentityModel;
+using IdentityServer4.Models;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Http.Authentication;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Security.Claims;
+
+namespace Fabric.Identity.API.Services
+{
+    public class ClaimsService
+    {
+        private readonly IAppConfiguration _appConfiguration;
+
+        public ClaimsService(
+            IAppConfiguration appConfiguration)
+        {
+            _appConfiguration = appConfiguration;
+        }
+
+        public ClaimsResult GenerateClaimsForIdentity(List<Claim> claims, AuthenticateInfo info, AuthorizationRequest context, string schemaItem)

```

Comment:
if you're passing in info, can't you get schemaItem from that?

Reply:
thought it would be messy to get it twice considering it requires doing this:  info.Properties.Items.FirstOrDefault(i => i.Key == "scheme").value

By just passing it into the function, the function doesnt need to know about how to get this.  Also reduces complexity.  Say if we have to refactor in the future, there is one spot and not two that must be modified.

Reply:
Ok, i think i got it.  I moved more of the initialization into the function and everything will be generated and returned in the ClaimsResult.  I think this makes the AccountController function cleaner and hides some of the magic of parsing everything.